src/main/java/seedu/address/logic/parser/LessonEditCommandParser.java

```diff
@@ -41,18 +44,12 @@ public LessonEditCommand parse(String args) throws ParseException {
                     PREFIX_SUBJECT, PREFIX_HOMEWORK, PREFIX_RATES, PREFIX_OUTSTANDING_FEES,
                     PREFIX_CANCEL, PREFIX_UNCANCEL);
 
-        Index[] indices;
-
-        try {
-            indices = ParserUtil.parseIndices(argMultimap.getPreamble());
-        } catch (ParseException pe) {
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
-                    LessonEditCommand.MESSAGE_USAGE), pe);
+        String[] preamble = ParserUtil.parsePreamble(argMultimap.getPreamble());
+        if (preamble.length != INDEX_ARGS_COUNT_STUDENT_LESSON) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, LessonEditCommand.MESSAGE_USAGE));
         }
-
-        assert indices.length == 2;
-        Index studentIndex = indices[0];
-        Index lessonIndex = indices[1];
+        Index studentIndex = ParserUtil.parseIndex(preamble[STUDENT_INDEX_ZERO_BASED]);
+        Index lessonIndex = ParserUtil.parseIndex(preamble[LESSON_INDEX_ZERO_BASED]);
```

Comment:
I don't think it's a big issue but perhaps you can do the catching and throwing of relevant error messages here like AB3 did since the 2 indices are separate here (same for ldelete?)